nptdms/reader.py

```diff
@@ -30,50 +30,86 @@ def __init__(self, tdms_file):
         :param tdms_file: Either the path to the tdms file to read
             as a string or pathlib.Path, or an already opened file.
         """
+        self._file_path = None
+        self._index_file_path = None
+        self._file = None
+        self._index_file = None
+
         self._segments = None
         self._prev_segment_objects = {}
         self.object_metadata = OrderedDict()
-        self._file_path = None
-        self._index_file_path = None
         self._segment_channel_offsets = {}
         self.tdms_version = None
 
         if hasattr(tdms_file, "read"):
             # Is a file
-            self._file = tdms_file
+            tag = tdms_file.read(4)
+            tdms_file.seek(0, os.SEEK_SET)
+
+            if tag == b"TDSh":
+                self._index_file = tdms_file
+            elif tag == b"TDSm":
+                self._file = tdms_file
+            else:
+                raise ValueError(
+                    f"File should either start with 'b`TDSh`' or 'b`TDSm`', submitted starts with '{tag}'.")
+
         else:
             # Is path to a file
-            self._file_path = str(tdms_file)
-            self._file = open(self._file_path, 'rb')
-            index_file_path = self._file_path + '_index'
-            if os.path.isfile(index_file_path):
-                self._index_file_path = index_file_path
+            source_path = str(tdms_file)
+            if source_path.endswith(".tdms_index"):
+                self._index_file_path = source_path
+                self._index_file = open(self._index_file_path, "rb")
+
+                filepath = source_path.split(".")[0] + ".tdms"
+                if os.path.isfile(filepath):
+                    self._file_path = filepath
+                    self._file = open(self._file_path, "rb")
```

Comment:
I thought this could be a use case to read data, when the source is a `.tdms_index`. With this you could be able to instantly read data with `TdmsChannel.read_data()` instead of rising an error stating you cannot read data from index file. What do you think about this approach?

Reply:
Right, that still feels a bit odd to me as the file path was passed as the index file rather than the data file so I wouldn't really expect to be able to read data. If someone wants to read data I think they should have to pass the file path rather than the index file path.

This behaviour might be surprising to users and it's an extra feature to maintain going forward that seems unnecessary.

Reply:
I see the point and removed it